packages/modules/devices/sma/sma_sunny_boy/inverter.py

```diff
@@ -66,7 +66,13 @@ def read(self) -> InverterState:
             power_total = self.tcp_client.read_holding_registers(40084, ModbusDataType.INT_16, unit=unit) * 10
             # Gesamtertrag (Wh) [E-Total] SF=2!
             energy = self.tcp_client.read_holding_registers(40094, ModbusDataType.UINT_32, unit=unit) * 100
+            # Power
             dc_power = self.tcp_client.read_holding_registers(40101, ModbusDataType.UINT_32, unit=unit) * 100
+            # Phasenstöme
+            current_L1 = self.tcp_client.read_holding_registers(30977, ModbusDataType.INT_32, unit=unit) * -1
+            current_L2 = self.tcp_client.read_holding_registers(30979, ModbusDataType.INT_32, unit=unit) * -1
+            current_L3 = self.tcp_client.read_holding_registers(30981, ModbusDataType.INT_32, unit=unit) * -1
+            currents = [current_L1 / 1000, current_L2 / 1000, current_L3 / 1000]
```

Comment:
```suggestion
            currents = [self.tcp_client.read_holding_registers(r, ModbusDataType.INT_32, unit=unit) / -1000
                                for r in (30977, 30979, 30981)]
```
So ist es etwas kompakter und nicht so viel redundanter Code.